src/Components/Components/test/RendererTest.cs

```diff
@@ -436,12 +436,15 @@ public void CanDispatchEventsToTopLevelComponents()
 
             // Act/Assert: Event can be fired
             var eventArgs = new UIEventArgs();
-            renderer.DispatchEvent(componentId, eventHandlerId, eventArgs);
+            var task = renderer.DispatchEventAsync(componentId, eventHandlerId, eventArgs);
             Assert.Same(eventArgs, receivedArgs);
+            Assert.Equal(TaskStatus.RanToCompletion, task.Status); // Completes synchronously
+
+            await task; // Does not throw
```

Comment:
This is a great way to express the flow, showing that the effects are synchronous plus the task itself doesn't fail. 👍